src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/pbs/PBSJobManager.groovy

```diff
@@ -730,70 +740,74 @@ class PBSJobManager extends ClusterJobManager<PBSCommand> {
+                TimeUnit usedWalltime = null
+                if (jobResult.get("resources_used.mem"))
+                    usedMem = new BufferValue(Integer.valueOf(jobResult.get("resources_used.mem").find(/(\d+)/)), BufferUnit.valueOf(jobResult.get("resources_used.mem")[-2]))
+                if (jobResult.get("resources_used.walltime"))
+                    usedWalltime = new TimeUnit(jobResult.get("resources_used.walltime"))
+
+                gj.setAskedResources(new ResourceSet(null, mem, cores, nodes, walltime, null, jobResult.get("queue"), additionalNodeFlag))
+                gj.setUsedResources(new ResourceSet(null, usedMem, null, null, usedWalltime, null, jobResult.get("queue"), null))
+
+                gj.setOutFile(jobResult.get("Output_Path"))
+                gj.setErrorFile(jobResult.get("Error_Path"))
+                gj.setUser(jobResult.get("euser"))
+                gj.setExecutionHosts(jobResult.get("exec_host"))
+                gj.setSubmissionHost(jobResult.get("submit_host"))
+                gj.setPriority(jobResult.get("Priority"))
+                gj.setUserGroup(jobResult.get("egroup"))
+                gj.setResourceReq(jobResult.get("submit_args"))
+                gj.setRunTime(jobResult.get("total_runtime") ? Duration.ofSeconds(Math.round(Double.parseDouble(jobResult.get("total_runtime"))), 0) : null)
+                gj.setCpuTime(jobResult.get("resources_used.cput") ? parseColonSeparatedHHMMSSDuration(jobResult.get("resources_used.cput")) : null)
+                gj.setServer(jobResult.get("server"))
+                gj.setUmask(jobResult.get("umask"))
+                gj.setJobState(parseJobState(jobResult.get("job_state")))
+                gj.setExitCode(jobResult.get("exit_status") ? Integer.valueOf(jobResult.get("exit_status")) : null)
+                gj.setAccount(jobResult.get("Account_Name"))
+                gj.setStartCount(jobResult.get("start_count") ? Integer.valueOf(jobResult.get("start_count")) : null)
+
+                DateTimeFormatter pbsDatePattern = DateTimeFormatter.ofPattern("EEE MMM ppd HH:mm:ss yyyy").withLocale(Locale.ENGLISH)
+                if (jobResult.get("qtime")) // The time that the job entered the current queue.
+                    gj.setSubmitTime(parseTime(jobResult.get("qtime")))
+                if (jobResult.get("start_time")) // The timepoint the job was started.
+                    gj.setStartTime(parseTime(jobResult.get("start_time")))
+                if (jobResult.get("comp_time"))  // The timepoint the job was completed.
+                    gj.setEndTime(parseTime(jobResult.get("comp_time")))
+                if (jobResult.get("etime"))  // The time that the job became eligible to run, i.e. in a queued state while residing in an execution queue.
+                    gj.setEligibleTime(parseTime(jobResult.get("etime")))
+
+                return queriedExtendedStates.put(it.getKey(), gj)
+
+            } catch (Exception ex) {
```

Comment:
Just querying a general exception and throwing a BEException looks a bit like: "I don't know what I am doing so I just do something". Do you know some of the possible exceptions? If so catch them and do proper messages.

Reply:
Well, it is an improvement over the situation before, without any exception handling. Now at least one learns, what input caused the problem. This is also the reason that I catch and rethrow, just to add context information.

The code is fragile and probably can throw NullPointerExceptions at one point or the other. Do you have any proposals, how I should deal with these?